File: src/2679.sum-in-a-matrix.2728/solution.ts
function matrixSum (nums: number[][]): number {
    for (const row of nums) {
        row.sort((a, b) => b - a);
    }
    let result = 0;
    for (let j = 0; j < nums[0].length; j++) {
        let max = nums[0][j];
        for (let i = 1; i < nums.length; i++) {
            max = Math.max(max, nums[i][j]);
        }
        result += max;
    }
    return result;
}

File: src/2679.sum-in-a-matrix.2728/solution.java
class Solution {
    public int matrixSum(int[][] nums) {
        for(int[] row : nums){
            Arrays.sort(row);
        }
        int result = 0;
        for(int j=0;j<nums[0].length;j++){
            int max = nums[0][j];
            for(int i=1;i<nums.length;i++){
                max = Math.max(nums[i][j],max);
            }
            result += max;
        }
        return result;
    }
}